base/abstractset.jl

```diff
@@ -476,6 +476,62 @@ function isdisjoint(a, b)
     _isdisjoint(a, b)
 end
 
+function isdisjoint(a::AbstractRange{T}, b::AbstractRange{T}) where T<:Union{Real,Char}
+    (isempty(a) || isempty(b)) && return true
+    fa, la = first(a), last(a)
+    fb, lb = first(b), last(b)
+    if step(a) < 0
+        fa, la = la, fa
+    end
+    if step(b) < 0
+        fb, lb = lb, fb
+    end
+    not_overlapping = (la < fb) | (lb < fa)
+    if not_overlapping
+        return true
+    else
+        return _overlapping_range_isdisjoint(a, b)
+    end
+end
+
+_overlapping_range_isdisjoint(a::AbstractRange{T}, b::AbstractRange{T}) where T = invoke(isdisjoint, Tuple{Any,Any}, a, b)
+
+function _overlapping_range_isdisjoint(a::AbstractRange{T}, b::AbstractRange{T}) where T<:Integer
+    m = gcd(step(a), step(b))
+    fa, fb = minimum(a), minimum(b)
+    if mod(fa - fb, m) == 0
+        shortrg, longrg = argmin(length, (a,b)), argmax(length, (a,b))
+        mis, mil = minimum(shortrg), minimum(longrg)
+        mxs, mxl = maximum(shortrg), maximum(longrg)
+
+        # number of elements in shortrg that overlap longrg
+        overlapping = length(shortrg)
+
+        if mis < mil
+            overlapping -= cld(mil - mis, abs(step(shortrg)))
+        end
+        if mxs > mxl
+            overlapping -= cld(mxs - mxl, abs(step(shortrg)))
+        end
+
+        if overlapping ≥ abs(step(longrg)/m)
```

Comment:
```suggestion
        if overlapping ≥ abs(cld(step(longrg), m))
```
We know that `m` divides `step(longrg)`, so there is no reason to introduce floating point numbers and the possibility or rounding error.